### ISSUE TITLE
Layout/LeadingCommentSpace issue

### ISSUE THREAD

**Issue by jwillemsen** [rubocop/rubocop#11642]
I found that `Layout/LeadingCommentSpace` detects and fixes some code it shouldn't fix. The problem appears when `#` is used within a character literal.

As example try the following code:

```
def foo
  @in = ''
  if @in != ?\' #'
    # do something
  end
end
```

## Expected behavior

This shouldn't give the Layout/LeadingCommentSpace warning

## Actual behavior

```
test.rb:3:17: C: [Correctable] Layout/LeadingCommentSpace: Missing space after #.
  if @in != ?\' #'
                ^^
```

## Steps to reproduce the problem

See the code above

## RuboCop version

```
$ [bundle exec] rubocop -V
1.46.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 3.0.0) [x86_64-linux]
```

**Comment by koic** [comment]
This does not appear to be a false positive.

`?\'` means `"'"`.

```ruby
?\' == "'" # => true
```

So this is the same as the evaluation below.

```ruby
"'" #'
```

The cop detects that `#'` comment should be `# '` as a role. Code highlighting also parses it that way:

```ruby
def foo
  @in = ''
  if @in != ?\' #'
    # do something
  end
end
```

**Comment by jwillemsen** [comment]
Yes, you are right, my apologies for the issue
